### PR TITLE
alertmanager: improve docs for updating tenant configuration

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

https://github.com/grafana/mimir/issues/6137 prompted me to document the behaviour of the `POST /api/v1/alerts` endpoint. This also adds an example of setting a template file via `mimirtool alertmanager load`, but now with a relative path allowed by the changes in https://github.com/grafana/mimir/pull/6138

#### Checklist

- [ ] ~~Tests updated~~
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/mimir/references/http-api/index.md [hunk 1]
```diff
@@ -976,10 +976,12 @@ Requires [authentication](#authentication).
 POST /api/v1/alerts
 ```
 
-Stores or updates the Alertmanager configuration for the authenticated tenant. The Alertmanager configuration is stored in the configured backend object storage.
+Stores or updates the Alertmanager configuration for the authenticated tenant. The Alertmanager configuration is stored in the configured backend object storage and shared between Alertmanager instances.
```
Comment (pstibrany) [inline docs/sources/mimir/references/http-api/index.md:979]: > and shared between Alertmanager instances.

That seems like unnecessary addition. I mean, how else can it work in Mimir's horizontally scalable Alertmanager?

File: docs/sources/mimir/references/http-api/index.md [hunk 2]
```diff
@@ -976,10 +976,12 @@ Requires [authentication](#authentication).
...
 
 This endpoint expects the Alertmanager **YAML** configuration in the request body and returns `201` on success.
 
+The names of the templates in `template_files` must be valid file names and not be relative or absolute paths. For example, both `/templates/my-template.tpl` and `./my-template.tpl` are invalid, whereas `my-template.tpl` is valid.
```
Comment (pstibrany) [inline docs/sources/mimir/references/http-api/index.md:983]: Technically speaking "valid file name" is also a relative path. Perhaps we can say:


```suggestion
The names of the templates in `template_files` must be valid file names and not contain any path separators. For example, both `/templates/my-template.tpl` and `./my-template.tpl` are invalid, whereas `my-template.tpl` is valid.
```

**Review by pstibrany — APPROVED** [review]
👍 (comments are just nitpicks, feel free to ignore)

File: docs/sources/mimir/references/http-api/index.md (hunk 1 above)
Comment (dimitarvdimitrov) [reply docs/sources/mimir/references/http-api/index.md:979]: that was probably just me but that sentence on its own felt like it's just leaking implementation details without a clear purpose (i.e. so what if the config ends up in object storage). But maybe I was too tired when writing this. I can revert this 
```suggestion
Stores or updates the Alertmanager configuration for the authenticated tenant. The Alertmanager configuration is stored in the configured backend object storage.
```

**Review by gotjosh — APPROVED** [review]

**Comment by dimitarvdimitrov** [comment]
merging this now; happy to address feedback from @grafana/docs-metrics after too

**Pull request merged**